src/osd/OSDMap.cc

```diff
@@ -2510,29 +2515,29 @@ void OSDMap::dump(Formatter *f) const
   }
   f->close_section();
   f->open_array_section("pg_temp");
-  for (const auto &elempg : *pg_temp) {
+  for (const auto pg : *pg_temp) {
     f->open_object_section("osds");
-    f->dump_stream("pgid") << elempg.first;
+    f->dump_stream("pgid") << pg.first;
     f->open_array_section("osds");
-    for (const auto &elems : elempg.second)
-      f->dump_int("osd", elems);
+    for (const auto osd : pg.second)
+      f->dump_int("osd", osd);
     f->close_section();
     f->close_section();
   }
   f->close_section();
 
   f->open_array_section("primary_temp");
-  for (const auto &elem : *primary_temp) {
-    f->dump_stream("pgid") << elem.first;
-    f->dump_int("osd", elem.second);
+  for (const auto pg : *primary_temp) {
```

Comment:
@liewegas My intention was to improve the Ceph code to the newer standard and better readable code. But the following is the comment from Kefu, which is also a valid point. So I request you to provide your suggestions on this, so that we could proceed with either merge or close this PR. Thanks.

> <kefu> Your change render it more difficult for us to identify when a change is introduced and why. fwiw, we use "git blame" to identify what a commit changes, or which commit a certain piece of code belongs. If we have lots of massive cleanup like this,  the commit history is more difficult to read for maintainers. But seems Sage is okay with it and that's why i tried to improve your PR before it gets merged. But i don't like it personally, it makes the code easier to read, but more difficult to maintain.